src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1631,12 +1631,29 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                 {
                     if (vkey == VK_ADD)
                     {
-                        // Alt '+' <number> is used to input Unicode code points.
-                        // Every time you press + it resets the entire state
-                        // in the original OS implementation as well.
-                        s.encoding = AltNumpadEncoding::Unicode;
-                        s.accumulator = 0;
-                        s.active = true;
+                        static const auto enabled = []() {
+                            wchar_t buffer[4]{};
+                            DWORD size = sizeof(buffer);
+                            RegGetValueW(
+                                HKEY_CURRENT_USER,
+                                L"Control Panel\\Input Method",
+                                L"EnableHexNumpad",
+                                RRF_RT_REG_SZ,
+                                nullptr,
+                                &buffer[0],
+                                &size);
+                            return size == 4 && memcmp(&buffer[0], L"1", 4) == 0;

```

Comment:
my gosh, it's actually a string? wild

Reply:
Should we use any of the `wil` registry helpers here? or is that just overhead.

things like `wil::reg::try_get_value<string>` and co

Reply:
I wouldn't. It's trivial code like this. I could replace the check with a `wcsnlen` though...